addons/stock_account/models/product.py

```diff
@@ -111,31 +111,48 @@ def write(self, vals):
             self.filtered(lambda p: p.cost_method != 'fifo')._change_standard_price(vals['standard_price'])
         return super(ProductProduct, self).write(vals)
 
-    @api.depends('stock_valuation_layer_ids')
-    @api.depends_context('to_date', 'company')
-    def _compute_value_svl(self):
-        """Compute totals of multiple svl related values"""
-        company_id = self.env.company
-        self.company_currency_id = company_id.currency_id
+    def _get_valuation_layer_group_domain(self):
+        company_id = self.env.company.id
         domain = [
             ('product_id', 'in', self.ids),
-            ('company_id', '=', company_id.id),
+            ('company_id', '=', company_id),
         ]
         if self.env.context.get('to_date'):
             to_date = fields.Datetime.to_datetime(self.env.context['to_date'])
             domain.append(('create_date', '<=', to_date))
-        groups = self.env['stock.valuation.layer']._read_group(domain, ['value:sum', 'quantity:sum'], ['product_id'])
+        return domain
+
+    def _get_valuation_layer_group_fields(self):
+        return ['value:sum', 'quantity:sum']
+
+    def _get_valuation_layer_groups(self):
+        domain = self._get_valuation_layer_group_domain()
+        group_fields = self._get_valuation_layer_group_fields()
+        return self.env['stock.valuation.layer'].read_group(domain, group_fields, ['product_id'])
```

Comment:
The original code uses `_read_group`, but your code replaces it with `read_group`. Was this change intentional? I'm asking because these methods have a bit different signatures and return the results in a slightly different way.

Reply:
No this was not on purpose. It is a transfer issue from v14.0

Reply:
If this was not on purpose, could you please change the code? :)

Reply:
Oh thx for the reminder. done.